zerver/lib/push_notifications.py

```diff
@@ -1351,17 +1351,11 @@ def handle_push_notification(user_profile_id: int, missed_message: Dict[str, Any
     if trigger == "private_message":
         trigger = NotificationTriggers.DIRECT_MESSAGE  # nocoverage
 
-    mentioned_user_group_name = None
     # mentioned_user_group_id will be None if the user is personally mentioned
     # regardless whether they are a member of the mentioned user group in the
     # message or not.
     mentioned_user_group_id = missed_message.get("mentioned_user_group_id")
-
-    if mentioned_user_group_id is not None:
-        user_group = UserGroup.objects.get(
-            id=mentioned_user_group_id, realm_id=user_profile.realm_id
-        )
-        mentioned_user_group_name = user_group.name
+    mentioned_user_group_name = get_mentioned_user_group_name([missed_message], user_profile)
```

Comment:
It looks like you kinda do this, but `mentioned_user_group_id` needs to be reset; the best option might be to just have the downstream code work with `mentioned_user_group.id` rather than these extra variables?

Reply:
Okay, made the changes.

> the best option might be to just have the downstream code work with mentioned_user_group.id rather than these extra variables?

`mentioned_user_group` can be None. So better would be to use these variables instead of if conditional when we use it.

Reply:
OK